gluefactory/models/extractors/joint_point_line_extractor.py

```diff
@@ -354,30 +354,36 @@ def _forward(self, data):
                 start_lines = sync_and_time()
             lines = []
             valid_lines = []
+            line_descs = []
+            line_indices = []
 
-            for df, af, kp in zip(line_distance_field, line_angle_field, rescaled_kp):
+            for df, af, kp, desc in zip(line_distance_field, line_angle_field, rescaled_kp, keypoint_descriptors):
                 line_data = {
                     "points": torch.clone(kp),
                     "distance_map": torch.clone(df),
                     "angle_map": torch.clone(af),
+                    "descriptors": torch.clone(desc),
                 }
-                img_line_indices = self.line_extractor(line_data)
-                # Line matchers expect the lines to be stored as line endpoints where line endpoint = idx of respective keypoint
-                img_lines = img_line_indices
-                if len(img_lines) == 0:
+                line_pred = self.line_extractor(line_data)
+                lines.append(line_pred["lines"])
+                line_descs.append(line_pred["line_descriptors"])
+                line_indices.append(line_pred["line_endpoint_indices"])
```

Comment:
are these indices in the old format? (2, #lines)

Reply:
They are (N,2). They are basically the old output of the line detection - indices of the keypoints that form the line endpoints.